analysis/analysis.py

```diff
@@ -157,6 +156,47 @@ def compute_reread_counts(student_data, question, context):
 
     return organized_data
 
+def unique_word_pattern(student_data):
+    """
+    Take the CSV File and analyze the readers' responses based on the two different contexts of the
+    2 questions (this is an ad/this is just a short story); analyze if the total number of unique
+    responses changed as more and more readers' responses are analyzed. Eventually prints out the
+    pattern along with the number of unique word in the text.
+    :param student_data
+    :return 2 lists of sets specifying unique responses at each point in time as a user
+    submits a response
+    """
+    response_ad = set()
+    response_story = set()
+    unique_word_tracker_ad = []
+    unique_word_tracker_story = []
+    question = "In three words or fewer, what is this text about?"
+
+    for data in student_data:
+        filtered_word_resp = filter_words(data['response'])
+        if data['question'] == question and data['context'] == "This is an ad.":
```

Comment:
Can the matching context also be passed in as an argument?

Reply:
Resolving this -- can be done in next step.